### PR TITLE
fix: ignore .webjs/routes.d.ts at any depth via **/.webjs/*

### LINKED ISSUE #365: dogfood: gitignore .webjs/routes.d.ts correctly in nested/monorepo apps
## Problem

`.webjs/routes.d.ts` is a per-machine generated route-types overlay that is meant to be gitignored (only `.webjs/vendor/` is committed; see `packages/cli/templates/.gitignore`). But in this monorepo three copies show up as UNTRACKED in `git status`:

```
?? docs/.webjs/
?? packages/ui/packages/website/.webjs/
?? website/.webjs/
```

Root cause: the gitignore pattern `.webjs/*` contains a slash, so git anchors it to the directory the `.gitignore` lives in (the repo root). It ignores `<root>/.webjs/*` but NOT nested apps' `website/.webjs/`, `docs/.webjs/`, `packages/ui/packages/website/.webjs/`. `git check-ignore` confirms none of the nested `routes.d.ts` are ignored, so the generated, per-machine file leaks into `git status` and risks being committed.

This also affects the SHIPPED scaffold gitignore (`packages/cli/templates/.gitignore`): a scaffolded app placed anywhere other than its own repo root, or any app-in-a-monorepo, has the same gap.

## Design / approach

Make the ignore match `.webjs/` at any depth while preserving the `vendor/` exception (the comment + `gitignore-vendor-not-ignored` lint rule must keep passing):

```
**/.webjs/*
!**/.webjs/vendor/
!**/.webjs/vendor/**
```

Apply to:
- the monorepo root `.gitignore` (fixes the three untracked dirs here)
- the shipped scaffold template `packages/cli/templates/.gitignore` (so scaffolded/nested apps are covered out of the box)
- `examples/blog/.gitignore` and any other in-repo app gitignore with the root-anchored form

Verify the `vendor/` negation still re-includes under the `**/` form (git re-include of a directory is allowed; the parent-exclusion footgun the existing comment warns about does not apply because `*`/`**/` match the children, not the `.webjs` dir itself).

## Acceptance criteria

- [ ] `git status` is clean of `.webjs/routes.d.ts` across all in-repo apps (root, website, docs, ui website)
- [ ] `git check-ignore` reports the nested `routes.d.ts` files as ignored
- [ ] `.webjs/vendor/importmap.json` is still tracked / un-ignored (the `gitignore-vendor-not-ignored` check still passes)
- [ ] The shipped scaffold template is fixed so a nested/monorepo scaffolded app does not leak the file
- [ ] Tests / lint cover the vendor-not-ignored invariant under the new pattern

### PR DESCRIPTION
**Pull request by vivek7405**
## Summary

Closes #365.

The gitignore pattern `.webjs/*` carries a slash, so git anchors it to the directory the `.gitignore` lives in. In this monorepo the nested in-repo apps (`website/`, `docs/`, `packages/ui/packages/website/`) leaked their generated `.webjs/routes.d.ts` into `git status`, because the root-anchored rule never reached them. `routes.d.ts` is a per-machine generated route-types overlay that is meant to be ignored (only `.webjs/vendor/` is committed), so this was a real bug, and the same gap shipped in the scaffold template (any app nested below its repo root had it).

The fix switches the gitignore to `**/.webjs/*` (plus the matching `**/` vendor negations), which matches `.webjs/` at any depth while still re-including the committed `.webjs/vendor/` pin at each depth (a re-included parent dir permits a child negation). Verified empirically before and after: nested `routes.d.ts` is now ignored at every depth, and the vendor pin stays tracked at both root and nested depths.

## What changed

- **Gitignores** switched to `**/.webjs/*` + `!**/.webjs/vendor/` + `!**/.webjs/vendor/**`:
  - root `.gitignore` (fixes the three untracked dirs in this repo)
  - `examples/blog/.gitignore`
  - `packages/cli/templates/.gitignore` (the shipped scaffold, so a nested/monorepo scaffolded app no longer leaks the file)
- **Docs:** `packages/cli/templates/AGENTS.md` (the "do NOT modify the `.webjs/` lines" block now documents the `**/` form and why it matters).
- **Check rule:** `gitignore-vendor-not-ignored` description + fix hint now recommend the depth-robust `**/.webjs/*`. The older anchored `.webjs/*` still passes the rule (the probe runs from the app root), so this is guidance only, not a behavior change.
- **Tests:** added a rule test that `**/.webjs/*` keeps the vendor pin un-ignored, plus a behavioral test that proves the fix end to end. The new pattern ignores a nested `routes.d.ts` while the old anchored `.webjs/*` leaks it (the counterfactual), and the vendor pin stays tracked at root and nested depths.

## Test plan / Definition of done

- [x] Unit coverage added with a counterfactual; full server suite green (2160 pass, 0 fail).
- [x] `git check-ignore` confirms the three nested `routes.d.ts` are now ignored; vendor pins stay tracked at root and nested depths; `git status` is clean of `.webjs/`.
- [x] Scaffold tests pass (12/12) with the new template `.gitignore`.
- [x] Dogfood boot: website `/` 200, docs `/` 307, ui-website `/` 200, no broken modulepreloads. Blog e2e N/A (this change touches no served or browser-wire code).
- [x] Docs: scaffold `AGENTS.md` updated. Root `AGENTS.md`, `agent-docs/`, and the docs site do not document the literal `.webjs/*` gitignore pattern, so nothing else drifts.
- Browser / e2e tests: **N/A** because this is a git-config and check-rule-string change with no DOM, hydration, or served-wire impact.
- `.dockerignore`: **aligned** for consistency. The repo-root `.dockerignore` already used `**/.webjs/*`, and the scaffold `AGENTS.md` note groups `.gitignore` and `.dockerignore`, so the scaffold template `.dockerignore` was updated to match (a nested Docker build context no longer ships the per-machine `.webjs` cache). This does not affect `git status`; it keeps the scaffold artifacts uniform.
- Version bump: **N/A** on this PR (assessed separately after merge per the release workflow).

### CODE REVIEW

File: packages/server/src/check.js [hunk 1]
```diff
@@ -754,9 +754,12 @@ export async function checkConventions(appDir) {
   // --- Rule: gitignore-vendor-not-ignored ---
   // The .gitignore pattern for .webjs/vendor/ is subtle: `.webjs/`
   // alone excludes the directory entirely and git can't re-include
-  // children of an excluded parent. The correct pattern is `.webjs/*`
-  // plus `!.webjs/vendor/` plus `!.webjs/vendor/**`. AI agents
-  // and human reviewers frequently "simplify" this back to `.webjs/`,
+  // children of an excluded parent. The correct pattern is `**/.webjs/*`
+  // plus `!**/.webjs/vendor/` plus `!**/.webjs/vendor/**` (the `**/`
+  // prefix ignores `.webjs/` at any depth so a nested app does not leak
```
Comment (vivek7405) [inline packages/server/src/check.js:759]: The description and fix hint moved to `**/.webjs/*`, but this implementation comment still said the correct pattern is `.webjs/*`, contradicting them within the same file.

**Review by vivek7405 — COMMENTED** [review]
First pass over the gitignore fix. The pattern semantics check out: `**/.webjs/*` ignores `routes.d.ts` at every depth while the `!**/.webjs/vendor/` negations keep the committed pin tracked at root and nested depths, and the behavioral test's counterfactual is a real proof (the old anchored `.webjs/*` genuinely leaks a nested `routes.d.ts`). Completeness is fine too: the three leaking apps have no own `.gitignore` for `.webjs`, so the root pattern governs them, and no docs-site page hard-codes the literal pattern. One in-file inconsistency, noted inline and already fixed.

File: packages/server/src/check.js (hunk 1 above)
Comment (vivek7405) [reply packages/server/src/check.js:759]: Fixed in [`6545bd7`](https://github.com/webjsdev/webjs/commit/6545bd71bc5455b85ac205b1e56fd6132dbaa34c).

File: packages/cli/templates/.gitignore [hunk 2]
```diff
@@ -3,25 +3,32 @@ node_modules/
...
 # webjs / framework caches.
 # `.webjs/routes.d.ts` (the generated route-types overlay, regenerated per
-# machine by `webjs types` / `webjs dev`) is correctly ignored by `.webjs/*`.
+# machine by `webjs types` / `webjs dev`) is correctly ignored by `**/.webjs/*`.
 # `.webjs/vendor/` is the EXCEPTION: it holds the committed importmap
 # manifest (.webjs/vendor/importmap.json) and optionally the vendored
 # bundle files (after `webjs vendor pin --download`). Both ship to
 # production via source control so the server doesn't need
-# api.jspm.io reachable at boot. Pattern is `.webjs/*` ignored,
+# api.jspm.io reachable at boot. Pattern is `**/.webjs/*` ignored,
```
Comment (vivek7405) [inline packages/cli/templates/.gitignore:11]: This comment still described the old `.webjs/*` form while the rest of the PR (and the actual pattern lines below) moved to `**/.webjs/*`, so the file documented the pattern two ways.

File: test/repo-health/gitignore-webjs-depth.test.mjs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      // the app is a nested package (the leak this PR fixes).
+      assert.equal(
+        isIgnored(repo, '.webjs/routes.d.ts'),
+        true,
+        `${shipped}: root .webjs/routes.d.ts must be ignored`,
+      );
+      assert.equal(
+        isIgnored(repo, 'packages/site/.webjs/routes.d.ts'),
+        true,
+        `${shipped}: a NESTED .webjs/routes.d.ts must be ignored (the #365 leak)`,
```
Comment (vivek7405) [inline test/repo-health/gitignore-webjs-depth.test.mjs:70]: New guard: copies each shipped gitignore into a temp repo and asserts a nested routes.d.ts is ignored while the vendor pin stays tracked. Without this, reverting any shipped file to `.webjs/*` reintroduced #365 with no failing test (the vendor rule only probes root paths, where both forms pass).

**Review by vivek7405 — COMMENTED** [review]
Second pass, zoomed in on test efficacy and pattern-interaction edge cases. Interaction is solid (deeply nested vendor bundles ship, multi-level nested vendor dirs ship, nothing else in the ignore files collides). But I found a real test hole worth closing and a leftover doc inconsistency, both now fixed. The behavioral rule tests verified git's semantics with inline patterns yet nothing pinned the assertion to the SHIPPED gitignores, and the vendor rule only probes root paths where both forms pass, so a revert to anchored `.webjs/*` would have reintroduced the leak with a green suite. Added a repo-health test that copies each shipped gitignore into a throwaway repo and check-ignores a nested routes.d.ts (counterfactual-verified), plus a scaffold-output assertion.

File: packages/cli/templates/.gitignore (hunk 2 above)
Comment (vivek7405) [reply packages/cli/templates/.gitignore:11]: Fixed in d912ec1 (comment now reads **/.webjs/*).

File: test/repo-health/gitignore-webjs-depth.test.mjs (hunk 3 above)
Comment (vivek7405) [reply test/repo-health/gitignore-webjs-depth.test.mjs:70]: Added in d912ec1; counterfactual-verified (reverting a shipped gitignore to .webjs/* fails this test).

**Review by vivek7405 — COMMENTED** [review]
Third pass, cross-file consistency and the new test code itself. The five prose/comment surfaces (root + blog + template gitignores, scaffold AGENTS.md, check.js description/impl-comment/fix) now tell one **/.webjs/* story, the only remaining .webjs/* mentions are the explicit back-compat note and the generic broken-collapsed-pattern failure message. The new repo-health helper maps git check-ignore exit codes correctly, strips the leaked GIT_* env, and cleans up in finally. One redundant-but-loose assertion tightened.

File: test/scaffolds/scaffold-integration.test.js [hunk 4]
```diff
@@ -203,6 +203,25 @@ test('scaffoldApp full-stack: writes the canonical full-stack app layout', async
...
+    // nested below its repo root must not leak its generated
+    // .webjs/routes.d.ts. The depth-robust `**/.webjs/*` prefix is what
+    // distinguishes the fix from the old root-anchored `.webjs/*`.
+    // Anchor to a line start (multiline) so these match the ACTIVE rule
+    // lines, not the surrounding comment prose that also names the
+    // pattern. Without the anchor a revert of the real rule to `.webjs/*`
+    // would still pass while a stale comment kept the `**/` text.
+    assert.match(
+      gitignore,
+      /^\*\*\/\.webjs\/\*$/m,
```
Comment (vivek7405) [inline test/scaffolds/scaffold-integration.test.js:216]: These matched anywhere in the file, so the **/.webjs/* text in the comment prose satisfied them even with the active rule line reverted to .webjs/*. The repo-health test is the load-bearing guard, but this one should still assert the real rule line. Anchored to a line start in this commit.

Comment (vivek7405) [reply test/scaffolds/scaffold-integration.test.js:216]: Fixed in [`7c3bbce`](https://github.com/webjsdev/webjs/commit/7c3bbceda38ac12f31de0272019ca334ccc6b20e).

**Review by vivek7405 — COMMENTED** [review]
Holistic pass. The gitignore behavior, rule text, and both test files are correct and the shipped artifacts are genuinely guarded. One completeness gap: the scaffold template .dockerignore was still on the anchored .webjs/* while the repo-root .dockerignore already ships **/.webjs/* and this PR's AGENTS.md note groups the two files, so the template was the lone outlier. Aligned it and tightened the dockerignore assertion to the active line.

File: test/scaffolds/scaffold-integration.test.js [hunk 5]
```diff
@@ -176,8 +176,8 @@ test('scaffoldApp full-stack: writes the canonical full-stack app layout', async
     // .dockerignore must preserve the .webjs/vendor negation (parent
     // exclusion would silently drop the committed importmap).
     const dockerignore = readFileSync(join(appDir, '.dockerignore'), 'utf8');
-    assert.match(dockerignore, /!\.webjs\/vendor\//,
-      '.dockerignore keeps .webjs/vendor/ (committed importmap ships)');
+    assert.match(dockerignore, /^!\*\*\/\.webjs\/vendor\/$/m,
+      '.dockerignore keeps **/.webjs/vendor/ (committed importmap ships)');
```
Comment (vivek7405) [inline test/scaffolds/scaffold-integration.test.js:180]: Updated to the **/.webjs/vendor/ form (and line-anchored), matching the template .dockerignore now that it uses the depth-robust pattern like the gitignore.

Comment (vivek7405) [reply test/scaffolds/scaffold-integration.test.js:180]: Aligned in [`3c1caab`](https://github.com/webjsdev/webjs/commit/3c1caab29aa826abd8ba973958ed11b077825f68); the template .dockerignore now matches the root repo + the .gitignore.

**Review by vivek7405 — COMMENTED** [review]
Clean pass. Verified the scaffold .dockerignore now matches the repo-root .dockerignore and the .gitignore (Docker keeps the committed .webjs/vendor/ pin at the build-context root under **/.webjs/*), the line-anchored scaffold assertions and the repo-health test guard the shipped artifacts, and nothing reads .webjs/* as the recommended pattern anymore. No remaining issues.
